content/howto/extensibility/how-to-create-a-native-pluggable-widget.md

```diff
@@ -0,0 +1,1344 @@
+            expandIcon: this.renderIcon(expandIcon, defaultExpandIconGlyph),
+            style
+        };
+
+        return <WrappedGroupBox {...props}>{content}</WrappedGroupBox>;
+    }
+    ```
+
+12. Go to Mendix Studio Pro and select for the expand icon property a Euro sign.
+13. Run the app locally and check out your changes.
+
+You should notice that you can't see any icon in the header of the group box. The reason for this is that our background color and text color are both black. Remember that in the **defaultStyle** constant of our display component we defined that text of React Native components that get the **headerContent** style applied to it, should be white. However, we are not explicitly applying this style to our **Icon** component that we pass from our container component to the display component. You could argue to move the creation of the **Icon** component inside your display component, but this will go against the strict seperation of concerns related to the container and display component, since the **Icon** component is Mendix specific. Therefore, it should be part of the container component.
+
+Let's fix the issue by introducing a default style for our container component.
+
+1. Add the following **defaultStyle** constant outside the container component class:
+
+   ```tsx
+   const defaultStyle: CustomStyle = {
+     container: {},
+     header: {},
+     headerContent: {
+       color: "#FFF",
+       fontSize: 16
+     },
+     content: {}
+   };
+   ```
+
+2. Import the flattenStyles function:
+
+   ```tsx
+   import { Style, flattenStyles } from "./utils/common";
+   ```
+
+3. Add the following private attribute to the container component:
+
+   ```tsx
+   private readonly styles = flattenStyles(defaultStyle, this.props.style);
+   ```
+
+4. Adjust the **renderIcon** method, so that it returns an **Icon** component with a color and size defined:
+
+   ```tsx
+   private renderIcon = (toBeRenderedIcon: DynamicValue<NativeIcon> | undefined, glyph: string) => {
+        const nativeIcon: NativeIcon =
+            toBeRenderedIcon && toBeRenderedIcon.status === ValueStatus.Available
+                ? toBeRenderedIcon.value
+                : { type: "glyph", iconClass: glyph };
+
+        return (
+            <Icon color={this.styles.headerContent.color} icon={nativeIcon} size={this.styles.headerContent.fontSize} />
+        );
+    };
+   ```
+
+5. Check the app in the Make It Native app and you should see the icons in white and the correct size.
+
+### 3.5 Adding a collapsible property
+
+We are close to a fully featured group box. Though there are two more features that would be of high use for a Mendix developer: setting whether the group box should be collapsible and setting the initial state of being collapsed or not.
+
+1. Head to **src/GroupBox.xml** and add a **collapsible** property underneath the **content** property:
+
+   ```xml
```

Comment:
not sure if we need the full file... in those examples.

Reply:
Kept it for now to give the reader a good overview of the changes.